p4-16/spec/P4-16-spec.mdk

```diff
@@ -1639,6 +1642,60 @@ directions:
   including values for the directionless parameters. The directionless
   parameters in this case behave like `in` parameters.
   See Section [#sec-invoke-actions] for further details.
+- Default parameter values are only allowed for 'in' or direction-less parameters;
+  these values must evaluate to compile-time constants.
+
+### Optional parameters
+
+A parameter that is annotated with the `@optional` annotation is
+optional: the user may supply no value for that parameter in an
+invocation.  Optional parameters can only appear for arguments of:
+packages, extern functions, extern methods, and extern object
+constructors.  Optional parameters cannot have default values.  If a
+function type has both optional parameters and default values then it
```

Comment:
It is unclear to me if the term `function type` here is referring to all of the four types mentioned in the last few sentences, or it is referring to the extern `function`, or it is referring to some other `function`?